compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLog.java

```diff
@@ -240,6 +255,10 @@ private void reportDiagnostic(PackageID packageID, DiagnosticCode diagnosticCode
             return;
         }
 
+        if (errorCount > 1 && hasOtherTypeInArgs(args)) {
```

Comment:
Shall we add a comment here that we're not suppressing errors with `other` when it is the only error